### ISSUE TITLE
Stay-on-screen notifications can't be dismissed without clicking on them

### ISSUE THREAD

**Issue by tanderson-google** [linuxmint/Cinnamon#7229]
```
 * Cinnamon version (cinnamon --version)
Cinnamon 3.4.6
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
Debian Testing
 ```

**Issue**

Stay-on-screen notifications can't be dismissed without clicking on them

**Steps to reproduce**

1.  Launch Google Chrome (version 64 or later)
2. Navigate to https://tests.peter.sh/notification-generator/#requireInteraction=true;;action=3
3. Click on "Display the notification"

**Expected behaviour**

There should be a close button allowing me to dismiss the notification, but the only way to dismiss it is to click on the body of the notification, triggering an alert() in the browser.

**Comment by mtwebster** [comment]
We're discussing this a bit internally.

- Why is the notification marked 'resident'? (I assume that's what chrome is doing here.)  Is this some browser spec or just an arbitrary decision?  The same effect has usually been achieved by just not setting an expiration.

- I think it's probably Chrome's responsibility to provide a 'dismiss' button.  We could of course add one to all notifications, but we've never needed to do that, and I hate to do so now just over this bug.
- We implemented a notification applet originally to catch notifications, so they wouldn't persist on the desktop, but would still be reviewable/actionable later.

My personal opinion is notifications with the 'resident' hint should be treated the same way:  If the applet is enabled, the notification displays for a time, then moves to the tray, otherwise it can be persistent like it is now.

**Comment by mtwebster** [comment]
Sorry, I should have checked - 'req interaction' seems to be what makes it persist, and that's setting the notification as "Critical", which honestly I don't agree with.  We try to more or less adhere to this spec: https://developer.gnome.org/notification-spec.  I can't really think of many things coming from a browser that would be considered critical as it's defined there.

**Comment by tanderson-google** [comment]
Setting the urgency to Critical works around a different issue: https://github.com/linuxmint/Cinnamon/issues/7179

'req interaction' is supposed to make the notification never-timeout.  Chrome correctly sets the timeout to 0, as specified in the FDO notification spec, but the notification expires on Cinnamon after only a few seconds.  Setting the urgency to critical was the only way I found to workaround the issue.

**Comment by clefebvre** [comment]
We came to the conclusion that a little X close button was needed in notifications.

**Comment by itzexor** [comment]
The close button was implemented in #7336
